### PR TITLE
Add update methods for safari and general settings

### PR DESCRIPTION
**Pull request by imurchie**
Use `appium-support` plist methods, and add a way to update the settings in general, and the safari user settings in particular.

### CODE REVIEW

File: lib/settings.js [hunk 1]
```diff
@@ -55,6 +51,14 @@ function plistPaths (sim, identifier) {
   return paths;
 }
 
+async function updateSettings (sim, plist, updates) {
+  log.debug(`Updating settings for ${plist}`);
+  let paths = plistPaths(sim, plist);
+  for (let path of paths) {
+    await update(path, updates);
```
Comment (jlipps) [inline lib/settings.js]: can these run in parallel? or is that too dangerous

File: lib/settings.js [hunk 2]
```diff
@@ -118,17 +123,56 @@ async function updateLocationSettings (sim, bundleId, authorized) {
...
+    newUserSettings.safariAllowPopups = settingSet.WebKitJavaScriptCanOpenWindowsAutomatically;
+  }
+  if (_.has(settingSet, 'WarnAboutFraudulentWebsites')) {
+    newUserSettings.safariForceFraudWarning = !settingSet.WarnAboutFraudulentWebsites;
+  }
+  if (_.size(newUserSettings) > 0) {
+    log.debug('Updating Safari user settings');
+    let userSettingsPerFile = {};
+    let curUserSettings = await readSettings(sim, 'userSettings');
+    // _.each(curUserSettings, function (userSettingSet, file) {
```
Comment (jlipps) [inline lib/settings.js]: commented code here and below

File: lib/settings.js [hunk 3]
```diff
@@ -118,17 +123,56 @@ async function updateLocationSettings (sim, bundleId, authorized) {
...
+    for (let [file, userSettingSet] of _.pairs(curUserSettings)) {
+      if (!_.has(userSettingSet, 'restrictedBool')) {
+        userSettingSet.restrictedBool = {};
```
Comment (jlipps) [inline lib/settings.js]: might want a comment above here explaining that theres this weird restrictedBool deal that is a magic thing for some reason

File: test/unit/settings-specs.js [hunk 4]
```diff
@@ -49,12 +49,12 @@ describe('settings', () => {
 
     it('should update a plist', async () => {
       let originalData = await read(tmpPlist);
-      originalData[0]['com.apple.locationd.bundle-/System/Library/PrivateFrameworks/Parsec.framework']
+      originalData['com.apple.locationd.bundle-/System/Library/PrivateFrameworks/Parsec.framework']
```
Comment (jlipps) [inline test/unit/settings-specs.js]: i realize this was duplicated before, but maybe this should be a constant string?

File: test/unit/settings-specs.js [hunk 5]
```diff
@@ -106,17 +104,18 @@ describe('settings', () => {
...
-        expect(data[0][weirdLocKey]).to.not.exist;
+        expect(data['com.apple.mobilesafari']).to.not.exist;
+        expect(data[weirdLocKey]).to.not.exist;
       });
 
       it('should update', async () => {
         await updateLocationSettings(sim, 'com.apple.mobilesafari', true);
 
         let finalData = await read(realClientFile);
+        console.log(finalData);
```
Comment (jlipps) [inline test/unit/settings-specs.js]: console.log

File: test/unit/settings-specs.js [hunk 6]
```diff
@@ -148,9 +147,10 @@ describe('settings', () => {
 
         for (let file of realCacheFiles) {
           let finalData = await read(file);
-          finalData[0]['com.apple.mobilesafari'].should.exist;
-          finalData[0]['com.apple.mobilesafari'].LastFenceActivityTimestamp.should.equal(412122103.232983);
-          finalData[0]['com.apple.mobilesafari'].CleanShutdown.should.be.true;
+          console.log(finalData);
```
Comment (jlipps) [inline test/unit/settings-specs.js]: console.log

**Comment by imurchie** [comment]
Updated, added handling of safari settings.

File: test/unit/settings-specs.js [hunk 7]
```diff
@@ -3,43 +3,38 @@
...
 
 chai.should();
 let expect = chai.expect;
 chai.use(chaiAsPromised);
 
 describe('settings', () => {
+  let sim;
+  before(() => {
+    // create a simulator object that returns our fixture directory
+    sim = sim = new SimulatorXcode6();
```
Comment (jlipps) [inline test/unit/settings-specs.js]: `sim = sim = `?

File: test/unit/settings-specs.js [hunk 8]
```diff
@@ -62,71 +57,66 @@ describe('settings', () => {
...
+      await fs.unlink(clientFile);
+      for (let file of cacheFiles) {
         await fs.unlink(file);
       }
     });
 
     describe('client plist', () => {
       let data;
-      let weirdLocKey = 'com.apple.locationd.bundle-/System/Library/' +
+      const weirdLocKey = 'com.apple.locationd.bundle-/System/Library/' +
```
Comment (jlipps) [inline test/unit/settings-specs.js]: line up the lines?

**Comment by jlipps** [comment]
looking great. made two small comments

**Comment by imurchie** [comment]
Addressed.

**Comment by jlipps** [comment]
LGTM!

**Pull request merged**